### PR TITLE
auto runs prettier and tests on precommit

### PR DESCRIPTION
**Pull request by snissn**
added precommit hooks that

1. only allows the commit to land if the solidity tests pass
2. automatically runs prettier solidity code formatter on commit

### CODE REVIEW

File: src/DealClient.sol [hunk 1]
```diff
@@ -1,27 +1,33 @@
...
+import {BytesCBOR} from "@zondax/filecoin-solidity/contracts/v0.8/cbor/BytesCbor.sol";
+import {BigNumbers} from "@zondax/filecoin-solidity/contracts/v0.8/external/BigNumbers.sol";
+import {CBOR} from "@zondax/filecoin-solidity/contracts/v0.8/external/CBOR.sol";
+import {Misc} from "@zondax/filecoin-solidity/contracts/v0.8/utils/Misc.sol";
+import {FilAddresses} from "@zondax/filecoin-solidity/contracts/v0.8/utils/FilAddresses.sol";
+import {MarketDealNotifyParams, deserializeMarketDealNotifyParams, serializeDealProposal, deserializeDealProposal} from "./Types.sol";
 
 using CBOR for CBOR.CBORBuffer;
 
 contract MockMarket {
```
Comment (shrenujbansal) [inline src/DealClient.sol]: Could you also move this into the test file?

**Comment by shrenujbansal** [comment]
LGTM. Just one small comment

**Comment by snissn** [comment]
Thanks! I'm sorry though I'm not understanding your comment. What would you like moved to the test file? I am understanding that you're looking for the imports added to the test file? The test file imports the src file and doesn't need all of the src file's inputs afaik

**Comment by shrenujbansal** [comment]
I just meant moving the MockMarket into the test file but don't worry about it. I've done that in another PR as well

**Review by shrenujbansal — APPROVED** [review]

**Pull request merged**